### PR TITLE
plan: remove useless plan.

### PR DESCRIPTION
**Pull request by hanfei1991**
For query t1 left join t2 left join t3 order by t1.a limit 1, there will be a topn operator between two joins. That is a redundant operator.

@winoros @lamxTyler PTAL

### CODE REVIEW

File: plan/logical_plans.go [hunk 1]
```diff
@@ -321,13 +321,27 @@ type TopN struct {
 	ByItems []*ByItems
 	Offset  uint64
 	Count   uint64
+
+	partial bool
```
Comment (shenli) [inline plan/logical_plans.go]: Add comment for this.

File: plan/logical_plans.go [hunk 2]
```diff
@@ -321,13 +321,27 @@ type TopN struct {
...
+// Limit represents offset and limit plan.
+type Limit struct {
+	*basePlan
+	baseLogicalPlan
+	basePhysicalPlan
+
+	Offset uint64
+	Count  uint64
+
+	partial bool
```
Comment (shenli) [inline plan/logical_plans.go]: ditto

File: plan/topn_push_down.go [hunk 3]
```diff
@@ -33,14 +33,21 @@ func (s *baseLogicalPlan) pushDownTopN(topN *TopN) LogicalPlan {
...
 	}
 	if topN != nil {
-		return topN.setChild(p)
+		return topN.setChild(p, false)
 	}
 	return p
 }
 
-func (s *TopN) setChild(p LogicalPlan) LogicalPlan {
+func (s *TopN) setChild(p LogicalPlan, eliminable bool) LogicalPlan {
```
Comment (shenli) [inline plan/topn_push_down.go]: Add comment for the second argument.

File: plan/logical_plans.go [hunk 4]
```diff
@@ -321,13 +321,29 @@ type TopN struct {
 	ByItems []*ByItems
 	Offset  uint64
 	Count   uint64
+
+	// partial is ture if this topn is generated by push-down optimization.
```
Comment (lamxTyler) [inline plan/logical_plans.go]: `ture` -> `true`.

File: plan/logical_plans.go [hunk 5]
```diff
@@ -321,13 +321,29 @@ type TopN struct {
...
+// Limit represents offset and limit plan.
+type Limit struct {
+	*basePlan
+	baseLogicalPlan
+	basePhysicalPlan
+
+	Offset uint64
+	Count  uint64
+
+	// partial is ture if this topn is generated by push-down optimization.
```
Comment (lamxTyler) [inline plan/logical_plans.go]: ditto.

**Pull request merged**